src/databricks/labs/lsql/dashboards.py

```diff
@@ -724,6 +749,72 @@ def _get_query_widget_spec(fields: list[Field], *, frame: WidgetFrameSpec | None
         return spec
 
 
+@unique
+class FilterType(str, Enum):
+    """The filter widget type"""
+
+    DATE_RANGE_PICKER = "DATE_RANGE_PICKER"
+    MULTI_SELECT = "MULTI_SELECT"
+    NONE = "NONE"
+
+    def as_widget_spec(self):
+        widget_spec_mapping = {
+            "DATE_RANGE_PICKER": DateRangePickerSpec,
+            "MULTI_SELECT": MultiSelectSpec,
+        }
+        if self.name not in widget_spec_mapping:
+            raise ValueError(f"Can not convert to filter widget spec: {self}")
+        return widget_spec_mapping[self.name]
+
+
+@dataclass
+class FilterTile(Tile):
+    _position: Position = dataclasses.field(default_factory=lambda: Position(0, 0, 3, 1))
+    dataset_columns: list[tuple[str, str]] = dataclasses.field(default_factory=list)  # (column, dataset_name)
+
+    def validate(self) -> None:
+        """Validate the tile
+
+        Raises:
+            ValueError : If the tile is invalid.
+        """
+        super().validate()
+        if not self.metadata.is_filter():
+            raise ValueError(f"Tile is not a filter file: {self}")
+
+    def get_layouts(self) -> Iterable[Layout]:
+        """Get the layout(s) reflecting this tile in the dashboard."""
+        if not self.dataset_columns:
+            logger.warning(f"Filter tile for path {self.metadata.path} has no matching dataset columns")
+            return
+        config = json.loads(self.content)
+        filter_type = FilterType[config.get("type", "MULTI_SELECT").upper()]
+        control_encodings: list[ControlEncoding] = []
+        queries = []
+        for idx, (column, dataset_name) in enumerate(self.dataset_columns):
+            filter_fields = [
+                Field(name=column, expression=f"`{column}`"),
+                Field(name=f"{column}_associativity", expression="COUNT_IF(`associative_filter_predicate_group`)"),
+            ]
+            filter_query = Query(dataset_name=dataset_name, fields=filter_fields, disaggregated=False)
+            filter_named_query = NamedQuery(name=f"{filter_type}_{column}_{idx}", query=filter_query)
+            queries.append(filter_named_query)
+            control_encodings.append(ControlFieldEncoding(column, filter_named_query.name, display_name=column))
```

Comment:
could you move `ControlFrieldEncoding` to a variable first before appending it the list, like the `NamedQuery` above?